bytecode-compiler/bytecompiler.rewrite.md

```diff
@@ -520,7 +489,46 @@ assembly into a code object:
                                   self.filename, name, firstlineno, lnotab,
                                   self.scope.freevars, self.scope.cellvars)
 
-This code object is littered with fields:
+Note that we begin the traversal process by calling
+`self(t.body)` -- this is because `CodeGen` is a _callable
+object_ [ref]. This means when you invoke an instance `c` of CodeGen as if
+it is a function, Python will translate that into `c.call(your args)`.
+So what does `__call__` do on `CodeGen`?
+
+        def __call__(self, t):
+            if isinstance(t, list): return concat(map(self, t)) 
+            assembly = self.visit(t)
+            return SetLineNo(t.lineno) + assembly if hasattr(t, 'lineno') else assembly
```

Comment:
Is it safe to introduce `SetLineNo` in v0? I was confused as to whether this was born in v1 along with the 'fatter' assembly language layer, or whether this was something that existed in nascent form even in v0.

I pulled the definition of `__call__` way up here because the entry point to the visitor was a mystery to me for a while, and I wasn't sure if it was the visitor framework that required `__call__` to do the visiting, or if this was a convention we had implemented.